### DISCUSSION TITLE
[SUGGESTION] Sound emitter logic block

### DISCUSSION THREAD

**Discussion by Rubaniac** [tsunamayo/Starship-EVO#3943] [Suggestions]
After the RGB logic block was added I had an idea for something very usefull in my opinion. At the time I was working on a system for my star trek alerts, and I thought it would be cool if we could have them with sound effects aswell!

I was thinking about a **logic block**, wich would play mp3 files from a gamefile that we put the mp3 files into. It would play them in a loop until a switch hooked up to it deactivates as a dedicated loop feature (like in the rgb block), or iniates once when a button signal is input into the block.

This would allow players to make their own sound effects for guns, doors, alarms, ambiant noise (engine sounds, systems etc)
and so much more, in addition to the already existing sound effects in the game.

A **radius control** and **volume control** would be handy. To avoid sound effect spamming a "disable sound block effects" button in the menu could always be implemented for multiplayer situations.
Blueprints would have to save the mp3 file aswell.

I hope something like this can eventually be added, thank you in advance for the consideration.

~Rubaniac

P.S
Do you like nutella on your crêpes?

**Comment by TIKIRobo** [comment]
I agree that something like this would be super cool but i think that there is room for greifing with these (think earape) and so there should be a built in max volume/decibel limit on what the block can play. I also think it would be cool if we could change the range of the block (by use of a menu similar to that of the gravity gens, but better, to select an area that the music plays in) so that we could have music that plays only in a club or something like that

**Comment by ZachZent** [comment]
I semi agree with a sound block, but at least for now it would probably be better to have a limited library of sounds. Mostly alarms for red alert and such, maybe in the future a sort of loud speaker so you can broadcast your voice to the entire ship.

**Comment by Garrett-C** [comment]
Perhaps someting that has a UI that lets you input notes etc could be a good compromise between inporting sounds and being stuck with fixed sounds. Would allow some nice variety of things (so not all ships have the same alarm sound) but isn't quite as out there as being able to just import sounds. 

Would need a variety of instruments etc as well.

**Comment by cloudrambler** [comment]
It could be set up like a synthesizer. You have a library of basic (non-offensive) sound components and a range of sliders to modulate those sounds. Variables like: pitch, speed, fade, warble, intensity, etc. could be used to make the sound seem unique. Then for more complex noises, multiple sounds could be linked to the same trigger, or to a logic sequencer (if we get one).

**Comment by tsunamayo** [comment]
I see, thats a good point but honestly if it comes it wont be soon... Cheers
